sonar-text-plugin/src/test/java/org/sonar/plugins/secrets/api/AuxiliaryMatcherTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * SonarQube Text Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.secrets.api;
+
+import java.util.List;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.sonar.plugins.secrets.configuration.model.matching.AuxiliaryPatternType;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class AuxiliaryMatcherTest {
+
+  @ParameterizedTest
+  @MethodSource
+  void auxiliaryPatternShouldBeDetectedAndCandidateSecretShouldNotBeRemoved(AuxiliaryPatternType patternType, String content,
+    String auxiliaryPattern) {
+    AuxiliaryMatcher auxiliaryMatcher = new AuxiliaryMatcher(
+      patternType, new PatternMatcher("\\b(" + auxiliaryPattern + ")\\b"));
+
+    PatternMatcher patternMatcher = new PatternMatcher("\\b(candidate secret)\\b");
```

Comment:
Maybe `"\\b(candidate secret)\\b"` could be extracted as it occurs in many tests